flux-middle/src/cache.rs

```diff
@@ -0,0 +1,55 @@
+use std::{fs::File, path::PathBuf};
+
+use flux_common::config::CONFIG;
+use rustc_hash::FxHashMap;
+use rustc_span::def_id::DefId;
+
+pub struct QueryCache {
+    entries: FxHashMap<String, u64>,
+}
+
+impl Default for QueryCache {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl QueryCache {
+    pub fn new() -> Self {
+        QueryCache { entries: FxHashMap::default() }
+    }
+
+    pub fn insert<Tag>(&mut self, did: DefId, constr_hash: u64) {
+        let str = crate::pretty::def_id_to_string(did);
+        self.entries.insert(str, constr_hash);
+    }
+
+    pub fn is_safe<Tag>(&self, did: DefId, constr_hash: u64) -> bool {
```

Comment:
Is there a reason to parameterize this by the tag?

Reply:
oops, vestige.